beacon-chain/blockchain/optimistic_sync.go

```diff
@@ -43,6 +43,12 @@ func (s *Service) notifyForkchoiceUpdate(ctx context.Context, headBlk block.Beac
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get finalized block")
 	}
+	if finalizedBlock == nil || finalizedBlock.IsNil() {
```

Comment:
This seems like a hack to patch the underlying real reason that is that we are not saving our finalized blocks to db

Reply:
The alternative seems to be worse since it will require saving during batch processing which defeats the purpose of batch processing. Resolving this.